Source/control.cpp

```diff
@@ -466,6 +466,40 @@ void AddPanelString(string_view str)
 		pnumlines++;
 }
 
+std::vector<string_view> splitStringViewByNewline(const string_view str, const char delim = '\n')

```

Comment:
Let's not allocate a vector on every frame.
You can use this implementation instead: https://github.com/OpenDingux/gmenu2x/blob/master/src/split_by_char.h

Reply:
I don't really understand what this is doing. It seems to me that I want to create a SplitByChar by constructor, get a SplitByCharIterator out of it by calling begin(), then I should be able to use the * operator with... nothing? That doesn't compile. It looks like I can move to the next string with ++, that makes sense, it's just actually pulling string_view objects out of this that doesn't.

Reply:
```c++
for (string_view s : SplitByChar(str, '\n')) {
}
```

Reply:
awesome